service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationValueQuantityMapper.java

```diff
@@ -0,0 +1,121 @@
+import org.hl7.fhir.dstu3.model.Quantity;
+
+public final class ObservationValueQuantityMapper {
+    private static final String UNITS_OF_MEASURE_SYSTEM = "http://unitsofmeasure.org";
+    private static final String UNCERTAINTY_EXTENSION = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-ValueApproximation-1";
+
+    private static final String NO_COMPARATOR_VALUE_TEMPLATE = "<value xsi:type=\"PQ\" value=\"%s\" unit=\"%s\"/>";
+    private static final String LESS_COMPARATOR_VALUE_TEMPLATE = "<value xsi:type=\"IVL_PQ\">"
+        + "<high value=\"%s\" unit=\"%s\" inclusive=\"false\"/></value>";
+    private static final String LESS_OR_EQUAL_COMPARATOR_VALUE_TEMPLATE = "<value xsi:type=\"IVL_PQ\">"
+        + "<high value=\"%s\" unit=\"%s\" inclusive=\"true\"/></value>";
+    private static final String GREATER_COMPARATOR_VALUE_TEMPLATE = "<value xsi:type=\"IVL_PQ\"><low value=\"%s\" "
+        + "unit=\"%s\" inclusive=\"false\"/></value>";
+    private static final String GREATER_OR_EQUAL_COMPARATOR_VALUE_TEMPLATE = "<value xsi:type=\"IVL_PQ\"><low value=\"%s\" "
+        + "unit=\"%s\" inclusive=\"true\"/></value>";
+    private static final String NO_COMPARATOR_NO_SYSTEM_VALUE_TEMPLATE = "<value xsi:type=\"PQ\" value=\"valueQuantity.value\" unit=\"1\">"
+        + "<translation value=\"valueQuantity.value\"><originalText>%s</originalText></translation></value>";
+    private static final String LESS_COMPARATOR_NO_SYSTEM_VALUE_TEMPLATE = "<value xsi:type=\"IVL_PQ\">"
+        + "<high value=\"%s\" unit=\"1\" inclusive=\"false\"><translation value=\"%s\">"
+        + "<originalText>%s</originalText></translation></high></value>";
+    private static final String LESS_OR_EQUAL_COMPARATOR_NO_SYSTEM_VALUE_TEMPLATE = "<value xsi:type=\"IVL_PQ\"><high "
+        + "value=\"%s\" unit=\"1\" inclusive=\"true\"><translation value=\"%s\">"
+        + "<originalText>%s</originalText></translation></high></value>";
+    private static final String GREATER_COMPARATOR_NO_SYSTEM_VALUE_TEMPLATE = "<value xsi:type=\"IVL_PQ\">"
+        + "<low value=\"%s\" unit=\"1\" inclusive=\"false\"><translation value=\"%s\">%s</originalText></translation></low></value>";
+    private static final String GREATER_OR_EQUAL_COMPARATOR_NO_SYSTEM_VALUE_TEMPLATE = "<value xsi:type=\"IVL_PQ\">"
+        + "<low value=\"%s\" unit=\"1\" inclusive=\"true\"><translation value=\"%s\">"
+        + "<originalText>%s</originalText></translation></low></value>";
+    private static final String UNCERTAINTY_CODE = "<uncertaintyCode code=\"U\" "
+        + "codeSystem=\"2.16.840.1.113883.5.1053\" displayName=\"Recorded as uncertain\"/>";
+
+    private ObservationValueQuantityMapper() {
+    }
+
+    public static String processQuantity(Quantity valueQuantity) {
+        String result = StringUtils.EMPTY;
+        if (isUncertaintyCodePresent(valueQuantity)) {
+            result += UNCERTAINTY_CODE;
+        }
+
+        if (!valueQuantity.hasComparator()) {
```

Comment:
(optional) possible to avoid the negation here

Reply:
I can see how the negation makes it easier to follow along with the written spec though ;)